### ISSUE TITLE
create tandem names via string concatenation

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/natural-selection#224]
When I started implementing this sim, we were trying to avoid creating tandem names via string concatenation, primarily because they are not searchable in code.   Consequently, I have more code than I'd like that looks like this example in AddMutationsPanel.js.

```js
    // A row for each gene
    const furRow = new Row( genePool.furGene, iconsAlignGroup, labelColumnAlignGroup, buttonColumnsAlignGroup, {
      tandem: options.tandem.createTandem( 'furRow' )
    } );
    const earsRow = new Row( genePool.earsGene, iconsAlignGroup, labelColumnAlignGroup, buttonColumnsAlignGroup, {
      tandem: options.tandem.createTandem( 'earsRow' )
    } );
    const teethRow = new Row( genePool.teethGene, iconsAlignGroup, labelColumnAlignGroup, buttonColumnsAlignGroup, {
      tandem: options.tandem.createTandem( 'teethRow' )
    } );
    const rows = [ furRow, earsRow, teethRow ];
```

What I'd really like to have done is:

```js
// A row for each gene
const rows = _.map( genePool.genes, gene => 
  new Row( gene, iconsAlignGroup, labelColumnAlignGroup, buttonColumnsAlignGroup, {
    tandem: options.tandem.createTandem( `${gene.tandemPrefix}Row` )
  } )
);
```

Changing this late in the game would be a big task, risky, and is probably not wise.  But what is the current policy on tandem names? Do they have to be searchable string literals? Or is it OK to create them programmatically when it prevents duplicated or boilerplate code?

**Comment by pixelzoom** [comment]
I see this was added to the PhET-iO Instrumentation Guide on 8/14/2020 in https://github.com/phetsims/phet-io/commit/6f83cc6ed9b00d6264d5a5397afd40e86389f4c6:

> In general, it is best to use full string literals when creating tandems. This makes it easier to search for the location of a given phetioID in code. That said, some cases yield better code using string concatentaion, for example ...

**Comment by samreid** [comment]
I leveraged the string concatenation in Gravity and Orbits, and it was very useful.  Up to you how you want to proceed for this sim.

**Comment by pixelzoom** [comment]
OK, thanks.  I'm leaning towards leaving things "as is" at this late date.  We'll see how that fares in code review.

**Comment by pixelzoom** [comment]
Here are the classes that feel "brute force" to me.  Again, I'm not sure if I will (or should) do anything about them at this point, but at least they are identified.

Model
- [ ] BunnyCounts (whiteFurCount, brownFurCount,…)
- [ ] BunnyVariety (fatherFurAllele, motherFurAllele,…)

View
- [ ] AddMutationsPanel (furRow, earsRow, teethRow)
- [ ] AllelesPanel (furRow, earsRow, teethRow)
- [ ] PopulationPanel (whiteFurCheckbox, brownFurCheckbox, …)
- [ ] ProportionsGraphNode (furColumn, earsColumn, teethColumn)
- [ ] ProportionsLegendNode (furLegendNode, earsLegendNode, teethLegendNode)

**Comment by pixelzoom** [comment]
I did quite a bit of generalization in the above commits, stuff that didn't seem too risky.  GenesVisibilityManager in particular is much nicer. 

Many of the remaining cases have to do with each gene or allele have a visibleProperty.  I experimented with generating these Properties by iterating over GenePool.genes, and using a Map to get the Property for a specific Gene.  But things always got very dicy, and I bailed.   

So... I think this is where I should leave it.  If I were to do it over, I would create many more things by iterating over GenePool.genes.
